### PR TITLE
update to game protocol 13.10

### PR DESCRIPTION
**Pull request by omarcopires**
### Pull Request Prelude

- [x] I have followed [proper The Forgotten Server code styling][code].
- [x] I have read and understood the [contribution guidelines][cont] before making this PR.
- [x] I am aware that this PR may be closed if the above-mentioned criteria are not fulfilled.

### Changes Proposed 
* Update protocol for login in version 13.10.
* Update item otb to 13.10.
* fixes the life and mana limit on the life bar.
* A new function in Lua to disable login screen music.
* Compatible client: https://github.com/omarcopires/qt-client/tree/protocol-update

Note: It is still necessary to create functions to add sounds to game actions, but with the protocol this can be easier.

[cont]: https://github.com/otland/forgottenserver/wiki/Contributing
[code]: https://github.com/otland/forgottenserver/wiki/TFS-Coding-Style-Guide

### CODE REVIEW

File: src/protocolgame.cpp [hunk 1]
```diff
@@ -3593,21 +3595,18 @@ void ProtocolGame::AddPlayerSkills(NetworkMessage& msg)
 		msg.add<uint16_t>(0);                                                   // base special skill
 	}
 
-	// fatal, dodge, momentum
-	msg.add<uint16_t>(0);
-	msg.add<uint16_t>(0);
-
-	msg.add<uint16_t>(0);
-	msg.add<uint16_t>(0);
+	msg.addByte(0); // mitigation(?)
```
Comment (Zbizu) [inline src/protocolgame.cpp:3598]: element magic levels
u8 list size
list entry structure: u8 client element id, u16 bonus element ml

File: data/scripts/network/cyclopedia_character.lua [hunk 2]
```diff
@@ -98,10 +100,10 @@ local function sendGeneralStats(self, msg)
...
 	msg:addByte(0) -- enable exp boost store button
 
-	msg:addU16(self:getHealth())
-	msg:addU16(self:getMaxHealth())
-	msg:addU16(self:getMana())
-	msg:addU16(self:getMaxMana())
+	msg:addU32(self:getHealth())
+	msg:addU32(self:getMaxHealth())
+	msg:addU32(self:getMana())
+	msg:addU32(self:getMaxMana())
```
Comment (ranisalt) [inline data/scripts/network/cyclopedia_character.lua:106]: lol @ cip making the game so easy this became a problem

File: src/protocolgame.cpp [hunk 3]
```diff
@@ -3534,8 +3536,8 @@ void ProtocolGame::AddPlayerStats(NetworkMessage& msg)
 {
 	msg.addByte(0xA0);
 
-	msg.add<uint16_t>(std::min<int32_t>(player->getHealth(), std::numeric_limits<uint16_t>::max()));
-	msg.add<uint16_t>(std::min<int32_t>(player->getMaxHealth(), std::numeric_limits<uint16_t>::max()));
+	msg.add<uint32_t>(std::min<int32_t>(player->getHealth(), std::numeric_limits<int32_t>::max()));
+	msg.add<uint32_t>(std::min<int32_t>(player->getMaxHealth(), std::numeric_limits<int32_t>::max()));
```
Comment (ranisalt) [inline src/protocolgame.cpp:3540]: You don't need the `min` here anymore

File: src/protocolgame.cpp [hunk 4]
```diff
@@ -3548,8 +3550,8 @@ void ProtocolGame::AddPlayerStats(NetworkMessage& msg)
 	msg.add<uint16_t>(0); // store exp bonus
 	msg.add<uint16_t>(player->getClientStaminaBonusDisplay());
 
-	msg.add<uint16_t>(std::min<int32_t>(player->getMana(), std::numeric_limits<uint16_t>::max()));
-	msg.add<uint16_t>(std::min<int32_t>(player->getMaxMana(), std::numeric_limits<uint16_t>::max()));
+	msg.add<uint32_t>(std::min<int32_t>(player->getMana(), std::numeric_limits<int32_t>::max()));
+	msg.add<uint32_t>(std::min<int32_t>(player->getMaxMana(), std::numeric_limits<int32_t>::max()));
```
Comment (ranisalt) [inline src/protocolgame.cpp:3554]: Same as above

File: src/protocolgame.cpp [hunk 5]
```diff
@@ -3593,21 +3595,21 @@ void ProtocolGame::AddPlayerSkills(NetworkMessage& msg)
...
+		msg.add<uint16_t>(0);
+		msg.add<uint16_t>(0);
+	}
 
 	// to do: bonus cap
-	msg.add<uint32_t>(player->hasFlag(PlayerFlag_HasInfiniteCapacity) ? 1000000
-	                                                                  : player->getCapacity()); // base + bonus capacity
-	msg.add<uint32_t>(player->hasFlag(PlayerFlag_HasInfiniteCapacity) ? 1000000
-	                                                                  : player->getCapacity()); // base capacity
+	uint32_t capacityValue = player->hasFlag(PlayerFlag_HasInfiniteCapacity) ? 1000000 : player->getCapacity();
```
Comment (ranisalt) [inline src/protocolgame.cpp:3610]:
```suggestion
	uint32_t capacityValue = player->hasFlag(PlayerFlag_HasInfiniteCapacity) ? std::numeric_limits<uint32_t>::max() : player->getCapacity();
```

Comment (omarcopires) [reply src/protocolgame.cpp:3610]: This is visual, we want this value to be displayed, as it always has been.

Comment (EPuncker) [reply src/protocolgame.cpp:3610]: if we don't cap it to 10k for gm characters, the client will display 273489820394820348

Comment (ranisalt) [reply src/protocolgame.cpp:3610]: More of a client bug but OK

Comment (omarcopires) [reply src/protocolgame.cpp:3610]: Do you want any more or can it be merged?

**Review by EPuncker — DISMISSED** [review]

**Review by EPuncker — APPROVED** [review]

**Comment by ranisalt** [comment]
Let's go :rocket:

**Comment by fernandorrmartins** [comment]
@omarcopires what client can i use to login? Otclient does not support 13.10

**Comment by ranisalt** [comment]
@fernandorrmartins if you mean github.com/edubart/otclient, development ceased a few years ago before Tibia 13 launched. You'll have to use github.com/mehah/otclient or another up-to-date fork.

**Comment by fernandorrmartins** [comment]
@ranisalt  thank you
Where can i find .dat and .spr to 13.10?

**Comment by Codinablack** [comment]
@fernandorrmartins there is no 13.10 .dat and .spr the main client moved passed using those for managing their assets and instead use protobuffs now. They are stored in the assets folder that is created from the installation. If you need help finding them, message me on otland. I have same name and icon on there.

**Comment by Zbizu** [comment]
> @omarcopires what client can i use to login? Otclient does not support 13.10

I remember opening a PR with protocol fixes for otc redemption for 13.20 or higher so it should be compatible with 13.10 as well.

**Comment by Codinablack** [comment]
> > @omarcopires what client can i use to login? Otclient does not support 13.10
> 
> I remember opening a PR with protocol fixes for otc redemption for 13.20 or higher so it should be compatible with 13.10 as well.

13.20 Is definitely supported, however 13.10 is not for some reason. If you try to just manually add it to the list of clients and use it anyway, you get packet errors. I have specific packet errors in my log if you care to help us figure out how to fix it, as mehah definitely said he will not be bothering with it.

**Comment by luanluciano93** [comment]
> > > @omarcopires what client can i use to login? Otclient does not support 13.10
> > 
> > 
> > I remember opening a PR with protocol fixes for otc redemption for 13.20 or higher so it should be compatible with 13.10 as well.
> 
> 13.20 Is definitely supported, however 13.10 is not for some reason. If you try to just manually add it to the list of clients and use it anyway, you get packet errors. I have specific packet errors in my log if you care to help us figure out how to fix it, as mehah definitely said he will not be bothering with it.
send me the errors I created a PR for you with version 13.10.
discord: luanluciano93

**Pull request merged**